website/docs/core-library/compilation-options.md

```diff
@@ -0,0 +1,79 @@
+---
+id: compilation-options
+title: Compilation options
+sidebar_label: Compilation options
+slug: /core-library/compilation-options
+---
+
+A number of options in CMake can be set when compiling the H3 core library. These are relevant for building
+different ways of using and testing the library and do not effect the underlying algorithms.
+
+When configuring with CMake, an option may be specified using `-D` on the command line, like so for setting
+`CMAKE_BUILD_TYPE` to `Release`:
+
+```bash
+cmake .. -DCMAKE_BUILD_TYPE=Release
+```
+
+## BUILD_ALLOC_TESTS
+
+Whether to build the parts of the [test suite](./testing) that exercise the [H3_ALLOC_PREFIX](./custom-alloc) feature.
```

Comment:
- Should we have examples?
- How do you set a boolean here?

Reply:
I'm not sure if it would be helpful to have examples under individual options. Maybe overall configurations that are common?

Booleans are set using CMake's ON/OFF strings. https://cmake.org/cmake/help/latest/command/set.html